main.cpp

```diff
@@ -7,42 +7,371 @@
 using namespace std;
 
 class Figure {
-public:
-	Figure() {}
-	virtual bool check_turn() { return true; }
+	int row, col;
+	int row_turn, col_turn;
+	char color;
+public:	
+	Figure(int st_row, int st_col, char st_color) {
+		row = st_row;
+		col = st_col;
+		color = st_color;
+		row_turn = col_turn = 0;
+	}
+	virtual bool check_turn(int dst_row, int dst_col, char dst_color) {
+	/*	if (dst_row < col) col_turn = -col_turn;
+		if (dst_row < row) row_turn = -row_turn;
+		for (int i = col_turn, j = row_turn; col_turn + i < dst_col; i += col_turn, j += row_turn) {
+			if (field[row + j][col + i]->name != "0") return 0;
+		}
+		return 1;*/
+		return true;
+		// TODO: make check based on row_turn, col_turn
+	}
+	char get_color() const {
+		return color;
+	}
+	char get_row() const {
+		return row;
+	}
+	char get_col() const {
+		return col;
+	}
+	void change_row(int row_n) {
+		row = row_n;
+	}
+	void change_col(int col_n)  {
+		col = col_n;
+	}
+	virtual char to_char() const = 0;
 };
 
 class Pawn : public Figure {
 public:
-	bool check_turn() {
-		// ...
+	Pawn(int st_row, int st_col, char st_color) : Figure(st_row, st_col, st_color) {
+		// TODO: define row_turn, col_turn
+	}
+	bool check_turn(int dst_row, int dst_col, char dst_color) {
+		if (get_col() == dst_col && get_row() == dst_row) { cout << "You do not move" << endl; return 0; }
```

Comment:
I personally really don't like one-line if statements, and I think that it is generally considered bad code style. Can you expand all your if's on multiple lines? And I also prefer to always have {} even if the body of the if is one line, I think it makes code clearer.